From 210ab8ade47199aa81fcd2ea50472551a66daeeb Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Mon, 19 Oct 2020 15:09:47 +0700
Subject: [PATCH 359/445] ARM: dts: r8a7742-iwg21d-q7: Add LCD panel support

Add support for EDT ETM0700G0DH6 panel on the Renesas R8A7742 (RZ/G1H)
board.

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7.dts | 55 +++++++++++++++++++++++++++++++++
 1 file changed, 55 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
index ebb7d81..45f67ab 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
@@ -132,6 +132,56 @@
 		regulator-max-microvolt = <3300000>;
 		regulator-always-on;
 	};
+
+	vcc_lvds: regulator@4 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "panel vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	lcd_backlight: backlight {
+		compatible = "pwm-backlight";
+
+		pwms = <&tpu 2 5000000 GPIO_ACTIVE_HIGH>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <7>;
+		power-supply = <&vcc_lvds>;
+		pinctrl-0 = <&backlight_pins>;
+		pinctrl-names = "default";
+	};
+
+	panel_lvds: panel@0 {
+		compatible = "edt,etm0700g0dh6", "simple-panel";
+
+		backlight = <&lcd_backlight>;
+		power-supply = <&vcc_lvds>;
+
+		port {
+		    #address-cells = <1>;
+		    #size-cells = <0>;
+		    reg = <0>;
+		    panel_lvds_input: endpoint {
+			    remote-endpoint = <&du_out_lvds0>;
+		    };
+		};
+
+		panel-timing {
+		    clock-frequency = <33260>;
+		    hactive = <800>;
+		    vactive = <480>;
+
+		    hfront-porch = <40>;
+		    hback-porch = <88>;
+		    hsync-len = <128>;
+
+		    vback-porch = <33>;
+		    vfront-porch = <10>;
+		    vsync-len = <2>;
+		};
+	};
 };
 
 &avb {
@@ -262,6 +312,11 @@
 		function = "can1";
 	};
 
+	backlight_pins: backlight {
+		groups = "tpu0_to2";
+		function = "tpu0";
+	};
+
 	buzzer_pins: buzzer {
 		groups = "tpu0_to0";
 		function = "tpu0";
-- 
2.7.4

